src/cascadia/Remoting/Monarch.cpp

```diff
@@ -855,7 +922,10 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
     Windows::Foundation::Collections::IVectorView<PeasantInfo> Monarch::GetPeasantInfos()
     {
         std::vector<PeasantInfo> names;
-        names.reserve(_peasants.size());
+        {
+            std::shared_lock lock{ _peasantsMutex };
+            names.reserve(_peasants.size());
+        }

```

Comment:
BTW This makes me certain that we should create a thread safe proxy class for the Monarch in the future.
(Basically wrap the public methods of the class and make them thread-safe. The actual MonarchImpl class can then continue pretending threads don't exist.)

Reply:
There is something to say about the simplicity of just throwing a `std::unique_lock lock {_mutex}` at the top of everything. There is a minor concern about event handlers that call back up to the monarch but that is a solvable problem too. :shrug:

Reply:
Yeah if you consider how we got event handlers, it's clear that recursive mutexes are the best solution here. You already have me entirely convinced about that. 😄
In either case I suppose that's something for a different time.